src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -436,9 +451,9 @@ def _instantiate_image_prediction(self, image: np.ndarray, prediction: Classific
         return ImageClassificationPrediction(image=image, prediction=prediction, class_names=self.class_names)
 
     def _combine_image_prediction_to_images(
-        self, images_predictions: Iterable[ImageClassificationPrediction], n_images: Optional[int] = None
+        self, images_predictions: Iterable[ImageClassificationPrediction], n_images: Optional[int] = None, show_progress_bar: bool = True
     ) -> ImagesClassificationPrediction:
-        if n_images is not None and n_images == 1:
+        if (n_images is not None and n_images == 1) or not show_progress_bar:
```

Comment:
So here you've introduced a bug in my opinion.
By adding `or not show_progress_bar:` to the condition `if (n_images is not None and n_images == 1)` you are forcing code to go `images_predictions = [next(iter(images_predictions))]` path if progress bar is disabled.
But this path would process only one element of the sequence, not all of them.
What you should be doing instead is this:
`images_predictions = [image_predictions for image_predictions in tqdm(images_predictions, total=n_images, desc="Predicting Images", disable=not show_progress_bar)]`

Btw it is a good idea to add a test to `tests/unit_tests/test_predict.py` which would take any detection model and run predict on few images with show_progress_bar=False and ensure than number of predictions matches number of input images.